pkg/controller/migmigration/dvm.go

```diff
@@ -197,3 +221,34 @@ func (t *Task) getDirectVolumeClaimList() *[]migapi.PVCToMigrate {
 	}
 	return nil
 }
+
+func (t *Task) deleteDirectVolumeMigrationResources() error {
+
+	// fetch the DVM
+	dvm, err := t.getDirectVolumeMigration()
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+
+	if dvm != nil {
+		// fetch the corresponding DVMP instance
+		dvmp, err := t.getDirectVolumeMigrationProgressForDVM(dvm)
```

Comment:
This logic may be unnecessary. I believe we set an owner ref on the DVMP to the DVM. This means we should be able to just delete the DVM and rely on the garbage collector to get the dvmps

Reply:
Thanks for the tip @dymurray , I have updated the code just to delete the DVM and DIM, this surely cascades the delete all to all the subsequent DVMPs and DISMs, I tested this for 2 PVs and 2 IS in a migplan.